scripts/world_bank/worldbank.py

```diff
@@ -0,0 +1,268 @@
+
+    # Only retain unique list 
+    country_df = country_df[country_df['IndicatorCode'].isin(indicators_to_keep)]
+
+    # Map country codes to MCF nodes
+    country_df['CountryNode'] = f"E:WorldBank->E{index + 1}"
+    country_mcf += COUNTRY_MCF.replace("{X}", str(index + 1)).replace("{ISO_CODE}", country_code)
+
+    # Add to total dataframe
+    worldbank_dataframe = worldbank_dataframe.append(country_df)
+
+  # Map indicator codes to unique Statistical Variable 
+  worldbank_dataframe['StatisticalVariable']= worldbank_dataframe['IndicatorCode'].apply(lambda code: f"WorldBank/{code.replace('.', '_')}")
+  worldbank_dataframe = worldbank_dataframe.rename({'year', 'Year'}, axis=1)
+
+  return worldbank_dataframe, country_mcf
+
+def output_csv_and_tmcf_by_grouping(worldbank_dataframe, tmcfs_for_stat_vars, indicator_codes, country_mcf):
+  """ Outputs TMCFs and CSVs for each grouping of stat vars. 
+
+    Args:
+      worldbank_dataframe: Dataframe containing all indicators for all countries
+      tmcfs_for_stat_varss:  Array of tuples of templace MCF, columns on stat var observations, indicator codes for that template MCF
+      indicator_codes: Pandas dataframe with INDICATOR_CODES to include
+      country_mcf: MCF nodes for each country to prepend to each MCF file
+  """
+  # Only include a subset of columns in the final csv
+  output_csv = worldbank_dataframe[['StatisticalVariable', 'IndicatorCode', 'CountryNode', 'Year', 'Value']]
+
+  for index, enum in enumerate(tmcfs_for_stat_vars):
+    tmcf, stat_var_obs_cols_to_include_in_csv, stat_vars = enum
+
+    if len(stat_vars) != 0:
+      with open(f"WorldBank_{index}.tmcf", 'w', newline='') as f_out:
+        f_out.write(tmcf)
+
+      matching_csv = output_csv[output_csv['IndicatorCode'].isin(stat_vars)]
+
+      # Includes IndicatorCode by default
+      if len(stat_var_obs_cols_to_include_in_csv) > 1:
+        matching_csv = pd.merge(matching_csv, indicator_codes[stat_var_obs_cols_to_include_in_csv], on="IndicatorCode")
+      matching_csv.to_csv(f"WorldBank_{index}.csv", index=False)
+
+def main(argv):
+  # Load StatVar generation list
+  indicator_codes = pd.read_csv("WorldBankIndicators.csv")
+
+  # Generate stat vars
+  with open("WorldBank_StatisticalVariables.mcf", "w+") as f_out:
+    for _, row in indicator_codes.iterrows():
+      f_out.write(build_stat_vars_from_indicator_list(row))
+
+  # Create template MCFs for each grouping of stat vars
+  tmcfs_for_stat_vars = group_stat_vars_by_observation_properties(indicator_codes)
+
+  # Download data for all countries
+  worldbank_countries = pd.read_csv("drive/My Drive/WorldBank/WorldBankCountries.csv")
+  worldbank_dataframe, country_mcf = download_indicator_data(worldbank_countries, indicator_codes)
+
+  # Output final CSVs and variables
+  output_csv_and_tmcf_by_grouping(worldbank_dataframe, tmcfs_for_stat_vars, indicator_codes, country_mcf)
+
+if __name__ == '__main__':
+  app.run(main)
+
```

Comment:
Let's eliminate blank lines at the end of the file.

Reply:
Changed to a single blank line at the end per PEP8 guidelines.